presentation/matrix_view.py

```diff
@@ -203,53 +208,71 @@ def get_thresholds(self, metric: MetricType) -> Thresholds:
         return self._config.packet_loss
 
     @staticmethod
-    def get_metric_value(metric: MetricType, cell: MeshColumn) -> MetricValue:
+    def get_metric_value(metric: MetricType, health: HealthItem) -> MetricValue:
         if metric == MetricType.LATENCY:
-            return cell.latency_millisec.value
+            return health.latency_millisec
         if metric == MetricType.JITTER:
-            return cell.jitter_millisec.value
-        return cell.packet_loss_percent.value
+            return health.jitter_millisec
+        return health.packet_loss_percent
 
     @classmethod
     def make_figure_annotations(cls, mesh: MeshResults, metric: MetricType) -> List[Dict]:
         annotations: List[Dict] = []
-        for row in reversed(mesh.rows):
-            for col in row.columns:
-                from_agent = mesh.agents.get_by_id(row.agent_id)
-                to_agent = mesh.agents.get_by_id(col.agent_id)
-                text = cls.get_text(metric, mesh.connection(from_agent.id, to_agent.id))
+        for from_id in reversed(mesh.connection_matrix.agent_ids):
+            for to_id in mesh.connection_matrix.agent_ids:
+                if from_id == to_id:
+                    continue
+                from_agent = mesh.agents.get_by_id(from_id)
+                to_agent = mesh.agents.get_by_id(to_id)
+                health = mesh.connection(from_agent.id, to_agent.id).latest_measurement
+                text = cls.format_health(metric, health)
                 annotations.append(
-                    dict(
-                        showarrow=False,
-                        text=text,
-                        xref="x",
-                        yref="y",
-                        x=to_agent.alias,
-                        y=from_agent.alias,
-                    )
+                    dict(showarrow=False, text=text, xref="x", yref="y", x=to_agent.alias, y=from_agent.alias)
                 )
         return annotations
 
     @staticmethod
-    def get_text(metric: MetricType, cell: MeshColumn) -> str:
+    def format_health(metric: MetricType, health: Optional[HealthItem], include_unit: bool = False) -> str:
+        not_available = "N/A"
+
+        if not health:
+            return not_available
+
         if metric == MetricType.LATENCY:
-            return f"{(cell.latency_millisec.value):.2f}"
+            if math.isnan(health.latency_millisec):
+                return not_available
+            result = f"{(health.latency_millisec):.2f}"
+            if include_unit:
```

Comment:
The string representing the metric unit needs to be more tightly coupled with the value. It would be a great use-case for (again) having `Metric` class with attributes `metric_type: MetricType, value: MetricValue, unit: str`. We can also store the unit in `HealthItem` and have a method to get it `get_unit(metric_type: MetricType)`.

Look how simple this code could be, if `HealtItem` returned value and unit based on `MetricType`:

```
if math.isnan(health.get_value(metric)):
   return not_available
else:
   return "{:.2f}{}".format(health.get_value(metric), health.get_unit(metric) if include_unit else "")
```

all 3 metrics in 4 lines and no need for modification, in the unlikely case of adding metrics in the future.

Reply:
This looks nice, thanks!